cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/azure/azure-dev/cli/azd/pkg/tools/docker"
 )
 
-type dockerCredentials struct {
+type DockerCredentials struct {
```

Comment:
it would be worth it to make this read-only (thread-safe).  

Specially as it becomes an exported type.

Reply:
Sorry, @vhvb1989, I'm not really sure I understand what you're asking for here.  How would this change?  Do you mean not exporting these values and instead adding methods like:

func (dc *DockerCredentials) Username() string { ... }

So they can't be set after use?

Reply:
> Do you mean not exporting these values and instead adding methods lik..

That's one strategy, yes. But it is currently a `nit` , just based on the idea of a future where this struct would be referenced by more than one thread. You don't have to do it for this PR